From b67321f0372ccc907667e500992bf9ce24f1502a Mon Sep 17 00:00:00 2001
From: Gabor Juhos <juhosg@openwrt.org>
Date: Tue, 1 Oct 2013 15:40:08 +0200
Subject: [PATCH 44/44] rt2x00: rt2800lib: fix VGC setup for RT3883

 - use the rt2800_write_bbp_with_rx_chain function for 'BBP 66'
   register programming,
 - use correct default VGC values,
 - use the correct adjustment value,

References:
  RT3883_ChipAGCAdjust in chips/rt3883.c

Signed-off-by: Gabor Juhos <juhosg@openwrt.org>
---
 drivers/net/wireless/rt2x00/rt2800lib.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/rt2x00/rt2800lib.c
+++ b/drivers/net/wireless/rt2x00/rt2800lib.c
@@ -4723,7 +4723,8 @@ static u8 rt2800_get_default_vgc(struct
 	} else { /* 5GHZ band */
 		if (rt2x00_rt(rt2x00dev, RT3572))
 			vgc = 0x22 + (rt2x00dev->lna_gain * 5) / 3;
-		else if (rt2x00_rt(rt2x00dev, RT3593))
+		else if (rt2x00_rt(rt2x00dev, RT3593) ||
+			 rt2x00_rt(rt2x00dev, RT3883))
 			vgc = 0x20 + (rt2x00dev->lna_gain * 5) / 3;
 		else if (rt2x00_rt(rt2x00dev, RT5592))
 			vgc = 0x24 + (2 * rt2x00dev->lna_gain);
@@ -4743,7 +4744,8 @@ static inline void rt2800_set_vgc(struct
 {
 	if (qual->vgc_level != vgc_level) {
 		if (rt2x00_rt(rt2x00dev, RT3572) ||
-		    rt2x00_rt(rt2x00dev, RT3593)) {
+		    rt2x00_rt(rt2x00dev, RT3593) ||
+		    rt2x00_rt(rt2x00dev, RT3883)) {
 			rt2800_bbp_write_with_rx_chain(rt2x00dev, 66,
 						       vgc_level);
 		} else if (rt2x00_rt(rt2x00dev, RT5592)) {
@@ -4785,6 +4787,8 @@ void rt2800_link_tuner(struct rt2x00_dev
 			vgc += 0x20;
 		else
 			vgc += 0x10;
+	} else if (rt2x00_rt(rt2x00dev, RT3883) && qual->rssi > -65) {
+		vgc += 0x10;
 	} else if (rt2x00_rt(rt2x00dev, RT5592) && qual->rssi > -65) {
 		vgc += 0x20;
 	} else if (qual->rssi > -80) {
